### PR TITLE
New blog newsroom

### PR DESCRIPTION
**Pull request by kurtw**
Introduces the new Blog and Newsroom page types

## Additions

- BlogPage and NewsroomPage 
- Article Subheader

## Changes

- Changes some templates around
- Fixes some bugs
- Learn, DocDetail, Blog, Newsroom pages have an option to add an Article Subheader for blog introductory paragraphs
- Full Width Text now has a Media option for images

## Testing

- `cfgov/manage.py migrate` (make sure you have the local.py database settings configured, if not see Notes)
- Go under the blog page and create a new blog post!
 - Add Article Subheader to the Header streamfield
 - Add the media option in the Content streamfield
 - Add some sidefoot content
- Preview it, Save it, Publish, Check it out on the site!
- Repeat for NewsroomPage

## Review

- @kave 
- @jimmynotjim 
- @richaagarwal 
- @rosskarchner 

## Screenshots
screen shot 2016-04-20 at 5 28 24 pm

## Notes

- local.py settings for database
```
DATABASES = {
    'default': {
        'ENGINE': MYSQL_ENGINE,
        'NAME': os.environ.get('MYSQL_NAME'),
        'USER': os.environ.get('MYSQL_USER'),
        'PASSWORD': os.environ.get('MYSQL_PW', ''),
        'HOST': os.environ.get('MYSQL_HOST', ''),  # empty string == localhost
        'PORT': os.environ.get('MYSQL_PORT', ''),  # empty string == default
        'OPTIONS': {
               "init_command": "SET storage_engine=MYISAM",
        }   
    },
}
```

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/article.html [hunk 1]
```diff
@@ -24,11 +24,15 @@
 
    ========================================================================== #}
 
-{% import 'molecules/social-media.html' as social_media with context %}
+{%- import 'molecules/social-media.html' as social_media with context -%}
+{%- import 'macros/category-slug.html' as category_slug -%}
+{%- import 'macros/time.html' as time -%}
+{%- import 'molecules/social-media.html' as social_media with context -%}
```
Comment (kave) [inline cfgov/jinja2/v1/_includes/article.html]: duplicate

File: cfgov/jinja2/v1/_includes/molecules/text-introduction.html [hunk 2]
```diff
@@ -24,9 +24,17 @@
 
    ========================================================================== #}
 
-<h1>{{ value.heading }}</h1>
-<div class="lead-paragraph">{{ parse_links(value.intro) | safe }}</div>
-{{ parse_links(value.body) | safe }}
+{% if value.heading %}
+    <h1>{{ value.heading }}</h1>
+{% endif %}
+{% if value.intro.src %}
```
Comment (kave) [inline cfgov/jinja2/v1/_includes/molecules/text-introduction.html]: ? why .src

File: cfgov/jinja2/v1/_includes/organisms/full-width-text.html [hunk 3]
```diff
@@ -12,6 +12,8 @@
 
    ========================================================================== #}
 
+{%- import 'templates/render_block.html' as render_block with context -%}
```
Comment (kave) [inline cfgov/jinja2/v1/_includes/organisms/full-width-text.html]: unused import lets remove

**Comment by kave** [comment]
fixed
> Random blog post with categories assigned that aren’t showing in “eyebrow” above the title

**Comment by kave** [comment]
👍 after comments

**Comment by kave** [comment]
All acceptance tests pass

**Comment by schaferjh** [comment]
Our blogs would be better if they involved more dog photos.

File: cfgov/jinja2/v1/_includes/molecules/text-introduction.html (hunk 2 above)
Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/text-introduction.html]: nah, `{{ if value.intro }}` is always true.

**Comment by kurtw** [comment]
Gotta love Australian Shepards ❤️

**Pull request merged**